pkg/report/linux.go

```diff
@@ -623,14 +623,20 @@ func (ctx *linux) parseOpcodes(codeSlice string) (parsedOpcodes, error) {
 	}, nil
 }
 
-// decompileReportOpcodes detects the most meaningful "Code: " lines from the report, decompiles
+// decompileOpcodes detects the most meaningful "Code: " lines from the report, decompiles
 // them and appends a human-readable listing to the end of the report.
-func (ctx *linux) decompileReportOpcodes(report []byte) []byte {
+func (ctx *linux) decompileOpcodes(text []byte, report *Report) []byte {
+	if extractReportType(report) == Hang {
```

Comment:
Isn't `rep.Type` already filled at this stage? If yes, use it. `extractReportType` is only meant to initialize `rep.Type`.

Reply:
Changed that (though at the cost of getting the test a bit more complicated).

PTAL